### PR TITLE
Missing values autodist

### PR DESCRIPTION
**Pull request by gverbock**

### CODE REVIEW

**Comment by gverbock** [comment]
@Matgrb I made a first attempt to solve the issue. Could you let me know if this is what you had in mind? 
I still want to merge the "new" test file with the existing ones. DId a separate tests because the original tests failed when run locally.

File: probatus/stat_tests/distribution_statistics.py [hunk 1]
```diff
@@ -339,11 +340,16 @@ def compute(
...
+            
             dist = DistributionStatistics(statistical_test=stat_test, binning_strategy=bin_strat, bin_count=bins)
             try:
                 if suppress_warnings:
                     warnings.filterwarnings("ignore")
-                _ = dist.compute(df1[col], df2[col])
+                # Issue a warning if missing values are present in one of the two columns. These observations are removed
+                # in the calculations. 
+                if np.sum(df1[col].isna()) + np.sum(df2[col].isna()):
+                    warnings.warn(f"Missing values in column {col+1} have been removed")
```
Comment (Matgrb) [inline probatus/stat_tests/distribution_statistics.py:351]: Why is it col+1? Shouldn't it be col? 

I think also col might be a string right?

Comment (Matgrb) [inline probatus/stat_tests/distribution_statistics.py:351]: In current implementation if you run 5 different tests for col1, the missing values are removed 5 times and you get 5 warnings.

Maybe good to improve the for loop in line 331, into two nested for loops. The outter one for col names and the inner one for the rest. This way you can remove the nans only once per column.

File: tests/stat_tests/test_missing_values_autodist.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+import pytest
+
+from probatus.stat_tests import ks, psi, DistributionStatistics, AutoDist
+from sklearn.datasets import make_classification
+
+def test_missing_values_in_autodist():
+    """Test missing values have no impact in AutoDist functionality. """
+    # Create dummy dataframe
+    X, y = make_classification(10000, 10)
```
Comment (Matgrb) [inline tests/stat_tests/test_missing_values_autodist.py:13]: Let's make the sample much smaller, to improve the speed of the test. 50x5 is already sufficient.

File: tests/stat_tests/test_missing_values_autodist.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    X = pd.DataFrame(X)
+    # Split train and test
+    X_train = X.sample(8000)
```
Comment (Matgrb) [inline tests/stat_tests/test_missing_values_autodist.py:16]: Let's use train test split from sklearn. Using sample function you will get an overlap between the two.

File: tests/stat_tests/test_missing_values_autodist.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        .compute(pd.concat([X_train, X_na]), pd.concat([X_test, X_na])))
+        
+    # Compute the statistics withpout the missing values
+    no_missing = (
+        AutoDist(statistical_tests=["PSI","KS"], binning_strategies="SimpleBucketer", bin_count=10)
+        .compute(X_train, X_test))
+        
+    # Test the two set of results are identical
+    pd.testing.assert_frame_equal(with_missings, no_missing)
+
```
Comment (Matgrb) [inline tests/stat_tests/test_missing_values_autodist.py:33]: Let's also test if the warning is printed the right number of times.
```
    with pytest.warns(None) as record:
        report = autodist.compute(...)

    assert len(record) == N # N= correct number of warnings printed
```

**Review by Matgrb — CHANGES_REQUESTED** [review]
Looks good already, couple of minor comments.

File: probatus/stat_tests/distribution_statistics.py (hunk 1 above)
Comment (gverbock) [reply probatus/stat_tests/distribution_statistics.py]: Correct. I have change it in the last push

Comment (gverbock) [reply probatus/stat_tests/distribution_statistics.py]: I split the loop in two loops. Should be more efficient now

File: tests/stat_tests/test_missing_values_autodist.py (hunk 2 above)
Comment (gverbock) [reply tests/stat_tests/test_missing_values_autodist.py]: Used 50 and 500. 500 was needed when computing the number of warning as I got additional warning on empty bucket. I could also reduce it to 50 and use 2 buckets for the tests on the number of warnings.

File: tests/stat_tests/test_missing_values_autodist.py (hunk 3 above)
Comment (gverbock) [reply tests/stat_tests/test_missing_values_autodist.py]: Done

File: tests/stat_tests/test_missing_values_autodist.py (hunk 4 above)
Comment (gverbock) [reply tests/stat_tests/test_missing_values_autodist.py:33]: The test is implemented. Shall l merge the tetst_missing_values_autodist.py file into test_distribution_statistics.py?

**Comment by gverbock** [comment]
@Matgrb I believe I made the required changes. Just one or two small details left to agree on.

File: tests/stat_tests/test_missing_values_autodist.py (hunk 2 above)
Comment (Matgrb) [reply tests/stat_tests/test_missing_values_autodist.py]: Then let's not use make_classification, but construct a list ourselves:
```
[0, 1, 2, 3,...] # with 10 samples and one missing
```

This way each bucket should have one value, and we have full control over what is going on. No randomness in the test.

File: tests/stat_tests/test_missing_values_autodist.py (hunk 4 above)
Comment (Matgrb) [reply tests/stat_tests/test_missing_values_autodist.py:33]: Yes, please do that

**Comment by Matgrb** [comment]
Two last minor changes and we can merge. Great job!

File: tests/stat_tests/test_missing_values_autodist.py (hunk 2 above)
Comment (gverbock) [reply tests/stat_tests/test_missing_values_autodist.py]: Adjusted to have a reproducible test on small scale.

File: tests/stat_tests/test_missing_values_autodist.py (hunk 4 above)
Comment (gverbock) [reply tests/stat_tests/test_missing_values_autodist.py]: Done!

**Comment by gverbock** [comment]
@Matgrb I believe this is it. Please let me know if you would like to see other changes.

**Review by Matgrb — APPROVED** [review]
Great contribution, thanks!

**Pull request merged**